src/Raven.Server/Documents/Handlers/Processors/Attachments/AttachmentHandlerProcessorForBulkPostAttachment.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Attachments;
+using Raven.Client.Documents.Operations.Attachments;
+using Raven.Client.Extensions;
+using Raven.Server.Documents.Handlers.Processors.Attachments.Strategies;
+using Raven.Server.Documents.PeriodicBackup.DirectDownload;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Attachments
+{
+    internal sealed class AttachmentHandlerProcessorForBulkPostAttachment : AbstractAttachmentHandlerProcessorForBulkPostAttachment<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public AttachmentHandlerProcessorForBulkPostAttachment([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask GetAttachmentsAsync(DocumentsOperationContext context, BlittableJsonReaderArray attachments, AttachmentType type,
+            OperationCancelToken tcs)
+        {
+            var tasks = new List<Task<Stream>>();
+            var attachmentsStreams = new List<Stream>();
+            DirectFileDownloader downloader = null;
+            bool canDisposeReadTransaction = true;
+            using DocumentsTransaction tx = context.OpenReadTransaction();
+            var stream = new MemoryStream();
```

Comment:
`RecyclableMemoryStreamFactory` ?

Reply:
I refactored the code and used the `RequestHandler.ResponseBodyStream()` directly in the writer